client/gutenboarding/store/actions.ts

```diff
@@ -1,14 +1,25 @@
 /**
  * Internal dependencies
  */
-import { ActionType, SiteType } from './types';
+import { ActionType, SiteType, Vertical } from './types';
 
 export const setSiteType = ( siteType: SiteType ) => ( {
-	type: ActionType.SET_SITE_TYPE,
+	type: ActionType.SET_SITE_TYPE as const,
```

Comment:
Casting `as const` allows the type to be inferred as `ActionType.SET_SITE_TYPE` instead of `ActionType`, the enum type.